sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -1816,7 +1816,13 @@ class Analyzer(override val catalogManager: CatalogManager)
         aggs: Seq[Expression]): Expression = expr match {
       case ordinal @ UnresolvedOrdinal(index) =>
         if (index > 0 && index <= aggs.size) {
```

Comment:
We can use `withPosition` here so the error doesn't need a TreeNode field as a parameter.
```scala
withPosition(ordinal) {
  ...
  throw QueryCompilationErrors.groupByPositionRefersToAggregateFunctionError(index, ordinalExpr)
}
```

Reply:
@cloud-fan Do we need to unify all case to use `withPosition`?